devito/types/parallel.py

```diff
@@ -237,152 +240,6 @@ def _C_typedata(self):
     def locked_dimensions(self):
         return set().union(*[d._defines for d in self.dimensions])
 
-    # Pickling support
```

Comment:
note for reviewers: most of this code has been moved to `ir/support/syncs.py` (and partly rewritten)